### DISCUSSION TITLE
error with model FashionMnist need your help

### DISCUSSION THREAD

**Discussion by ShreshShaurya** [mrdbourke/pytorch-deep-learning#1263] [Q&A]
# Create a convolutional neural network
class FashionMNISTModelV2(nn.Module):
    """
    Model architecture copying TinyVGG from:
    https://poloclub.github.io/cnn-explainer/
    """
    def __init__(self, input_shape: int, hidden_units: int, output_shape: int):
        super().__init__()
        self.block_1 = nn.Sequential(
            nn.Conv2d(in_channels=input_shape,
                      out_channels=hidden_units,
                      kernel_size=3, # how big is the square that's going over the image?
                      stride=1, # default
                      padding=1),# options = "valid" (no padding) or "same" (output has same shape as input) or int for specific number
            nn.ReLU(),
            nn.Conv2d(in_channels=hidden_units,
                      out_channels=hidden_units,
                      kernel_size=3,
                      stride=1,
                      padding=1),
            nn.ReLU(),
            nn.MaxPool2d(kernel_size=2,
                         stride=2) # default stride value is same as kernel_size
        )
        self.block_2 = nn.Sequential(
            nn.Conv2d(hidden_units, hidden_units, 3, padding=1),
            nn.ReLU(),
            nn.Conv2d(hidden_units, hidden_units, 3, padding=1),
            nn.ReLU(),
            nn.MaxPool2d(2)
        )
        self.classifier = nn.Sequential(
            nn.Flatten(),
            # Where did this in_features shape come from?
            # It's because each layer of our network compresses and changes the shape of our input data.
            nn.Linear(in_features=hidden_units*7*7, #used a trick to calculate this
                      out_features=output_shape)
        )

    def forward(self, x: torch.Tensor):
        x = self.block_1(x)
        # print(x.shape)
        x = self.block_2(x)
        # print(x.shape)
        x = self.classifier(x)
        # print(x.shape)
        return x
        
<img width="587" height="390" alt="image" src="https://github.com/user-attachments/assets/fddcf6be-ce5d-4841-9100-7f68c2e6f318" />
What I don't understand is how the accuracy in eval is worse but during test and train is ok.
I am not able to pinpoint the error. Would love to know where I am doing wrong.
<img width="366" height="646" alt="image" src="https://github.com/user-attachments/assets/fcd14104-d067-406f-adfc-de87a4fdcdac" />

Also, it is just detecting pullover, I feel like I messed up the data somewhere but I don't understand how.
<img width="593" height="562" alt="image" src="https://github.com/user-attachments/assets/4dfbac51-c261-407c-839b-4394b268a4e7" />

**Comment by fawern** [comment]
@ShreshShaurya 
You’re predicting along the wrong dimension when you turn logits into labels.
It looks like you do argmax(dim=0), which compares across the batch. That makes every item in the batch get the same class (hence the single column in the confusion matrx, everything becomes Pullover).
logits = model(x)            # [batch, num_classes]
preds  = logits.argmax(dim=1)  # use dim=1 (classes), not dim=0

Don’t apply softmax before CrossEntropyLoss (it expects raw logits).
After switching to argmax(dim=1), the eval accuracy and confusion matrix should look normal.
